controller/contacts.js

```diff
@@ -6,8 +6,40 @@ const {
 } = require("../joi");
 
 const get = async (req, res, next) => {
+  const { page, limit, favorite } = req.query;
+  let paginatedContacts = [];
+  const { _id } = req.user;
   try {
-    const contacts = await service.getAllContacts();
+    const contacts = await service.getAllContacts(_id);
+    if (!page && !limit && favorite) {
+      const filtered = contacts.filter(
+        (contact) => String(contact.favorite) === favorite,
+      );
+      return res.json({ data: filtered, status: "success", code: 200 });
+    }
+    if (page && limit && !favorite) {
+      const start = (Number(page) - 1) * Number(limit);
```

Comment:
To tylko informacyjne bo serwer nie przestaje odpowiadać, ale warto takie parametry spróbować walidować przy zmianie ze stringa na liczbę, ponieważ:
Number('5') => 5, ale
Number('a') => NaN i to nam psuje slice